t/t1001-mf-priority.t

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+test_description='Test multi-factor priority plugin'
+
+. `dirname $0`/sharness.sh
+MULTI_FACTOR_PRIORITY=${FLUX_BUILD_DIR}/src/plugins/.libs/mf_priority.so
+# BULK_UPDATE=${FLUX_BUILD_DIR}/t/scripts/send_fake_payloads.py
+
+export TEST_UNDER_FLUX_NO_JOB_EXEC=y
+export TEST_UNDER_FLUX_SCHED_SIMPLE_MODE="limited=1"
+test_under_flux 1 job
+
+flux setattr log-stderr-level 1
+
+test_expect_success 'load multi-factor priority plugin' '
+        flux jobtap load -r .priority-default ${MULTI_FACTOR_PRIORITY}
+'
+
+test_expect_success 'list all loaded plugins' '
+        flux jobtap list -a > test.txt
+'
+
+test_expect_success 'make sure multi-factor priority plugin is loaded' '
+        test_cmp test.txt ${FLUX_BUILD_DIR}/t/expected/loaded_priority.txt
+'
+
+test_expect_success 'try to submit a job when user does not exist in DB' '
+        test_must_fail flux mini submit -n1 hostname > failure.out 2>&1 &&
+        test_debug "cat failure.out" &&
+        grep "user not found in flux-accounting DB" failure.out
+'
+
+test_expect_success 'create fake_payload.py' '
+cat <<-EOF >fake_payload.py
+import flux
+import pwd
+import getpass
+
+username = getpass.getuser()
+userid = pwd.getpwnam(username).pw_uid
+# create a JSON payload
+data = {"userid": str(userid), "bank": "account3", "fairshare": "0.45321"}
+flux.Flux().rpc("job-manager.mf_priority.rec_update", data).get()
+data = {"userid": str(userid), "bank": "account2", "fairshare": "0.11345"}
+flux.Flux().rpc("job-manager.mf_priority.rec_update", data).get()
+EOF
+'
+
+test_expect_success 'update plugin with sample test data' '
+        flux python fake_payload.py
+'
+
+test_expect_success 'submit a job with default urgency' '
+        jobid=$(flux mini submit --setattr=system.bank=account3 -n1 hostname) &&
+        flux job wait-event -f json $jobid priority | jq '.context.priority' > test_job1.txt &&
+        test_cmp test_job1.txt ${FLUX_BUILD_DIR}/t/expected/job_1_priority.txt
```

Comment:
Instead of storing expected output separately from the test under version control, it is probably easier and more maintainable to create the expected output within the test itself (you'll see the pattern quite often in flux-core tests), e.g. here:
```
test_expect_success 'submit a job with default urgency' '
￼        jobid=$(flux mini submit --setattr=system.bank=account3 -n1 hostname) &&
￼        flux job wait-event -f json $jobid priority | jq '.context.priority' > test_job1.txt &&
         cat >job1.expected <<-EOF
         4096
         EOF
￼        test_cmp job1.expected test_job1.txt
```

Also note that usage of `test_cmp` is `test_cmp expected actual`.

I would use this method for almost all tests unless the expected output is very large.

Reply:
Thanks for pointing this out. I've just pushed a change to use the above pattern to compare the test result to the expected value. This feels much cleaner than the approach I was originally using.